### PR TITLE
letsencrypt: 0.1.0 -> 0.4.0

### PR DESCRIPTION
**Pull request by zimbatm**
###### Things done:
- [x] Tested via `nix.useChroot`.
- [x] Built on platform(s): NixOS amd64
- [x] Tested compilation of all pkgs that depend on this change.
- [ ] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

An updated version of #12123.

When running it tries to access /etc/letsencrypt. Any ideas @benley ?

```
$ letsencrypt 
An unexpected error occurred:
OSError: [Errno 13] Permission denied: '/etc/letsencrypt'
Please see the logfile 'letsencrypt.log' for more details.
```

### CODE REVIEW

**Comment by benley** [comment]
That's just the default behaviour of letsencrypt, if I recall correctly.  You have to add some extra commandline args to get it to behave nicely in your homedir.

**Comment by zimbatm** [comment]
Hmm okay, that's not ideal. Everyone should be using simp_le really :p

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8443,7 +8443,7 @@ let
     let imported = import ../development/libraries/qt-5/5.5 { inherit pkgs; };
     in recurseIntoAttrs (imported.override (super: qt5LibsFun));
 
-  qt5 = qt54;
+  qt5 = qt55;
```
Comment (vcunat) [inline pkgs/top-level/all-packages.nix]: Are these all-packages.nix changes intended? They seem rather irrelevant.

Comment (zimbatm) [inline pkgs/top-level/all-packages.nix]: woops, good catch. Fixed now

**Comment by zimbatm** [comment]
Merging as it's better than before and working provided you set the config folder.

**Comment by zimbatm** [comment]
@benley oops, do you also get FAILing tests when installing letsencrypt ?

**Comment by benley** [comment]
to be honest I haven't actually used letsencrypt since I put together the initial PR, sorry :-P

**Comment by benley** [comment]
thanks for finishing the update work, to be sure

**Comment by globin** [comment]
simp_le broke with this update

**Comment by globin** [comment]
build log: https://hydra.mayflower.de/build/332745/nixlog/5

**Comment by zimbatm** [comment]
sorry, I'll fix it tonight

**Comment by zimbatm** [comment]
@globin ^ PR at #13655
